sign.go

```diff
@@ -398,12 +398,28 @@ func (m *SignMessage) UnmarshalCBOR(data []byte) error {
 // Notice: The COSE Sign API is EXPERIMENTAL and may be changed or removed in a
 // later release.
 func (m *SignMessage) Sign(rand io.Reader, external []byte, signers ...Signer) error {
+	return m.sign(rand, external, nil, signers)
+}
+
+// SignDetached signs a SignMessage using the provided signers corresponding to the
+// signatures.
+//
+// See `Signature.Sign()` for advanced signing scenarios.
+//
+// Reference: https://datatracker.ietf.org/doc/html/rfc8152#section-4.4
+//
+// # Experimental
+//
+// Notice: The COSE Sign API is EXPERIMENTAL and may be changed or removed in a
+// later release.
+func (m *SignMessage) SignDetached(rand io.Reader, external, detachedPayload []byte, signers ...Signer) error {
```

Comment:
I think that we should always return an error if `SignMessage.Payload` is not empty, instead of reusing it when `detachedPayload` is nil (which is also an error). If someone is using `SignDetached`, then the intent is clear, the payload should be the one passed in the parameter, not the one from the message,